clarity/contracts/arkadiko-claim-usda-yield-v1-1.clar

```diff
@@ -0,0 +1,163 @@
+  )
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-dao-owner)) (err ERR-NOT-AUTHORIZED))
+
+    (try! (return-usda (get usda claim-entry)))
+    (map-set claims { vault-id: (get to recipient) } { usda: (- (get usda claim-entry) (get usda recipient)) })
+    (ok true)
+  )
+)
+
+;; @desc Claim PoX yield and pay off debt
+;; @param vault-id; your vault ID
+;; @param reserve; active STX reserve
+;; @param coll-type; active collateral types contract
+;; @post boolean; returns true if claim was succesful
+(define-public (claim
+  (vault-id uint)
+  (reserve <vault-trait>)
+  (coll-type <collateral-types-trait>)
+)
+  (let (
+    (vault (contract-call? .arkadiko-vault-data-v1-1 get-vault-by-id vault-id))
+    (stacker-name (get stacker-name vault))
+    (claim-entry (get-claim-by-vault-id vault-id))
+    (sender tx-sender)
+  )
+    (asserts! (not (var-get claim-shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq tx-sender (get owner vault)) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of reserve) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "stx-reserve"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of coll-type) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "collateral-types"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (> (get usda claim-entry) u0) (err ERR-NOTHING-TO-CLAIM))
+
+    (try! (as-contract (contract-call? .usda-token transfer (get usda claim-entry) tx-sender sender none)))
+
+    (map-set claims { vault-id: vault-id } { usda: u0 })
+    (ok true)
+  )
+)
+
+;; @desc Claim PoX yield and pay off debt
+;; @param vault-id; your vault ID
+;; @param reserve; active STX reserve
+;; @param coll-type; active collateral types contract
+;; @post boolean; returns true if claim was succesful
+(define-public (claim-and-burn
+  (vault-id uint)
+  (reserve <vault-trait>)
+  (coll-type <collateral-types-trait>)
+)
+  (let (
+    (vault (contract-call? .arkadiko-vault-data-v1-1 get-vault-by-id vault-id))
+    (stacker-name (get stacker-name vault))
+    (claim-entry (get-claim-by-vault-id vault-id))
+    (sender tx-sender)
+  )
+    (asserts! (not (var-get claim-shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq tx-sender (get owner vault)) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of reserve) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "stx-reserve"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of coll-type) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "collateral-types"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (> (get usda claim-entry) u0) (err ERR-NOTHING-TO-CLAIM))
+
+    (try! (as-contract (contract-call? .usda-token transfer (get usda claim-entry) tx-sender sender none)))
+    (try! (contract-call? .arkadiko-freddie-v1-1 burn vault-id (get usda claim-entry) reserve .arkadiko-token coll-type))
```

Comment:
Q: If USDA to claim is higher than debt, the difference will remain in the contract? Can we check/calculate this excess amount in the contract?
(We could check vault debt and update the map accordingly instead of setting it to 0)

Reply:
@nieldeckx afaik if you burn less than the usda amount parameter you passed, the difference will stay in your wallet? need to add a test for this

Reply:
added a test, this should work as I commented